app/models/provider.rb

```diff
@@ -63,4 +63,20 @@ def refresh_ems(opts = {})
     end
     managers.flat_map { |manager| EmsRefresh.queue_refresh(manager, nil, opts) }
   end
+
+  def self.destroy_queue(ids)
+    find(Array.wrap(ids)).each(&:destroy_queue)
+  end
+
+  def destroy_queue
+    if managers.empty?
+      return destroy
+    end
+
+    _log.info("Queuing destroy of managers of provider: #{self.class.name} with id: #{id}")
+    managers.flat_map(&:destroy_queue)
```

Comment:
Why are you `flat_map`ping and then not using the return value... just use `.each`

Reply:
haha, it was before
using `.each` now. thanks!